dev-lang/julia/julia-1.9.4.ebuild

```diff
@@ -0,0 +1,218 @@
+
+EAPI=8
+
+MY_LLVM_V=14.0.6
+
+PYTHON_COMPAT=( python3_{9..11} )
+
+inherit check-reqs flag-o-matic optfeature pax-utils python-any-r1 toolchain-funcs
+
+DESCRIPTION="High-performance programming language for technical computing"
+HOMEPAGE="https://julialang.org/
+	https://github.com/JuliaLang/julia/"
+SRC_URI="
+	https://github.com/JuliaLang/julia/releases/download/v${PV}/${P}-full.tar.gz
+"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS=""
+#KEYWORDS="~amd64 ~x86"
+PROPERTIES="test_network"
+RESTRICT="test"
+
+RDEPEND="
+	>=dev-libs/libutf8proc-2.6.1:0=[-cjk]
+	>=dev-util/patchelf-0.13
+	>=net-libs/mbedtls-2.2
+	>=sci-mathematics/dsfmt-2.2.4
+	>=sys-libs/libunwind-1.1:0=
+	>=virtual/blas-3.6
+	app-arch/p7zip
+	app-misc/ca-certificates
+	dev-libs/gmp:0=
+	dev-libs/libgit2:0
+	dev-libs/mpfr:0=
+	net-misc/curl[http2,ssh]
+	sci-libs/amd:0=
+	sci-libs/arpack:0=
+	sci-libs/camd:0=
+	sci-libs/ccolamd:0=
+	sci-libs/cholmod:0=
+	sci-libs/colamd:0=
+	sci-libs/fftw:3.0=[threads]
+	sci-libs/openlibm:0=
+	sci-libs/openblas:0=
+	sci-libs/spqr:0=
+	sci-libs/umfpack:0=
+	sys-libs/zlib:0=
+	virtual/lapack
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	${PYTHON_DEPS}
+	dev-util/cmake
+	virtual/pkgconfig
+"
+
+CHECKREQS_DISK_BUILD="4G"
+# Avoid building ocaml binding with LLVM_ENABLE_BINDINGS=OFF #47862 (merged upstream but not yet for 1.9.4)
+PATCHES=(
+	"${FILESDIR}"/julia-1.8.0-llvm-no-bindings.patch
+	"${FILESDIR}"/julia-1.9.4-string-replace.patch
+	)
+QA_FLAGS_IGNORED+='usr/.*/julia/sys.so'  # Julia sysimage generated by bootstrapping.
+
+# Huge thanks to Arch Linux developers for the patches.
+archlinux_uri="https://raw.githubusercontent.com/archlinux/svntogit-community/packages/julia/trunk/"
+archlinux_patches=(
+)
+for archlinux_patch in ${archlinux_patches[@]} ; do
+	archlinux_patch_name="${PN}-1.8.0-${archlinux_patch}"
+	SRC_URI+="
+		${archlinux_uri}/${archlinux_patch}
+			-> ${archlinux_patch_name}
+	"
+	PATCHES+=( "${DISTDIR}/${archlinux_patch_name}" )
+done
+
+pkg_setup() {
+	check-reqs_pkg_setup
+	python-any-r1_pkg_setup
+}
+
+src_unpack() {
+	local -a tounpack=( ${A} )
+	# the main source tree, followed by deps
+	unpack "${tounpack[0]}"
+
+	mkdir -p "${S}/deps/srccache/"
+	local i
+	for i in "${tounpack[@]:1}"; do
+		cp "${DISTDIR}/${i}" "${S}/deps/srccache/${i#julia-}" || die
+	done
+
+	# Extract tarballs for patching.
+	# cd "${S}/deps/srccache/" || die
+	# tar xf llvm-julia-${MY_LLVM_V}-3.tar.gz || die
+}
+
+src_prepare() {
+	default
+
+	# Sledgehammer:
+	# - prevent fetching of bundled stuff in compile and install phase
+	# - respect CFLAGS
+	# - respect EPREFIX and Gentoo specific paths
+
+	sed -i \
+		-e "\|SHIPFLAGS :=|c\\SHIPFLAGS := ${CFLAGS}" \
+		Make.inc || die
+
+	sed -i \
+		-e "s|ar -rcs|$(tc-getAR) -rcs|g" \
+		src/Makefile || die
+
+	# disable doc install starting	git fetching
+	sed -i -e 's~install: $(build_depsbindir)/stringreplace $(BUILDROOT)/doc/_build/html/en/index.html~install: $(build_depsbindir)/stringreplace~' Makefile || die
```

Comment:
I think we can put this in a patch, can we? or wqill it get overwritten by Julia's makefile magic?

Reply:
A patch has been added instead.